caddyhttp/gzip/setup.go

```diff
@@ -119,3 +124,52 @@ func gzipParse(c *caddy.Controller) ([]Config, error) {
 
 	return configs, nil
 }
+
+// pool gzip.Writer according to compress level
+// so we can reuse allocations over time
+var (
+	writerPool             = map[int]*sync.Pool{}
```

Comment:
I don't think we need a lock here, we only write this map during initialization and then only read.